### DISCUSSION TITLE
conditional query prisma

### DISCUSSION THREAD

**Discussion by Joseamica** [prisma/prisma#16053] [Q&A]
I want to find the menus type provided by the restaurant (breakfast type) according to the time now, and the hours to display is on ShowingHours model

I created a Menu model
```
model Menu {
  id             String         @id @default(cuid())
  type           String?
  branchId       String
  branch         Branch         @relation(fields: [branchId], references: [id])
  menuCategories MenuCategory[]

  ShowingHours   ShowingHours? @relation(fields: [showingHoursId], references: [id])
  showingHoursId String?
}
```

and ShowingHours
```
model ShowingHours {
  id         String  @id @default(cuid())
  fromHour   Int?
  fromMinute Int?
  toHour     Int?
  toMinute   Int?
  allDay     Boolean
  menus      Menu[]
}
```

```
    let timeNow = new Date().getHours();
```

I want to make a condition that if the value of timeNow for example is 9 am, then query a menu that has a inbetween timenow hours from "fromHour" to "toHour"

My approach:
```
let menuType = await db.menu.findMany({
      // between fromhour tohour
     where: {
      branchId: branchId,
//I WOULD LIKE TO MAKE A CONDITION LIKE: if timeNow >= fromHour && timenow < toHour then fetch 
      ShowingHours: {
        OR: [
          {
            fromHour: {
              in: 
            }
          }
        ]
      }
      },
     
      
    })

```

**Comment by austincrim** [comment]
Hey @Joseamica 👋,

You should be able to use the [`gte`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#gte) and [`lte`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#lte) options for this:

```ts
let menuType = await db.menu.findMany({
  where: {
    branchId: branchId,
    ShowingHours: {
      fromHour: { gte: 9 },
      toHour: { lte: 12 } 
    }
  }
})
```

Let me know if that doesn't provide the expected results.

**Reply by vladi-stevanovic** [reply]
:wave: Hello @Joseamica, I’m going through pending GitHub Discussions to check if you need any help. Is this issue still open?

Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
